src/category_theory/sites/sheaf.lean

```diff
@@ -257,6 +257,20 @@ def Sheaf_to_presheaf : Sheaf J A ⥤ (Cᵒᵖ ⥤ A) :=
 instance : full (Sheaf_to_presheaf J A) := { preimage := λ X Y f, ⟨f⟩ }
 instance : faithful (Sheaf_to_presheaf J A) := {}
 
+instance Sheaf.hom.mono_of_presheaf_mono {F G : Sheaf J A} (f : F ⟶ G) [mono f.1] : mono f :=
+{ right_cancellation := λ P g h eq₀, Sheaf.hom.ext _ _ $ (cancel_mono f.1).mp $
+    (Sheaf.hom.ext_iff _ _).mp eq₀ }
+
+instance Sheaf.hom.epi_of_presheaf_epi {F G : Sheaf J A} (f : F ⟶ G) [epi f.1] : epi f :=
+{ left_cancellation := λ P g h eq₀, Sheaf.hom.ext _ _ $ (cancel_epi f.1).mp $
+    (Sheaf.hom.ext_iff _ _).mp eq₀ }
+
+instance Sheaf_to_presheaf_reflects_mono : (Sheaf_to_presheaf J A).reflects_monomorphisms :=
+infer_instance
+
+instance Sheaf_to_presheaf_reflects_epi : (Sheaf_to_presheaf J A).reflects_epimorphisms :=
+infer_instance
```

Comment:
```suggestion
instance Sheaf.hom.mono_of_presheaf_mono {F G : Sheaf J A} (f : F ⟶ G) [h : mono f.1] : mono f :=
(Sheaf_to_presheaf J A).mono_of_mono_map h

instance Sheaf.hom.epi_of_presheaf_epi {F G : Sheaf J A} (f : F ⟶ G) [h : epi f.1] : epi f :=
(Sheaf_to_presheaf J A).epi_of_epi_map h
```
I think this would be enough? Are the two instance needed to be restated? There should be a comment explaining the reason if they are needed.

Also, since there probably aren't many instances of the form `mono f.1` lying around, I think it would be more useful if `presheaf_mono_of_mono` was an instance instead?

Reply:
Indeed, I actually only need`presheaf_mono_of_mono` but I thought I should include these just for completeness.